ouroboros-consensus/ouroboros-consensus-test-infra/src/Test/ThreadNet/Network.hs

```diff
@@ -428,18 +436,41 @@ runThreadNetwork ThreadNetworkArgs
             forM_ edgeStatusVars $ \edgeStatusVar -> atomically $ do
               readTVar edgeStatusVar >>= check . (== EDown)
 
-            -- close the ChainDB
+            -- assuming nothing else is changing it, read the final chain
             let chainDB = getChainDB kernel
             finalChain <- ChainDB.toChain chainDB
 
-            pure (again, finalChain)
+            pure (again, finalChain)   -- end of the node's withRegistry
 
           atomically $ writeTVar vertexStatusVar $ VDown finalChain
 
           case again of
             Nothing                     -> pure ()
             Just (s', pInfo', nr', rs') -> loop s' pInfo' nr' rs'
 
+    -- | Persistently attempt to add the given transactions to the mempool
+    -- every time the ledger slot changes, even if successful!
+    --
+    -- If we add the transaction and then the mempools discards it for some
+    -- reason, this thread will add it again.
+    --
+    forkTxs0
```

Comment:
This could do with a one or two line explanation of what the purpose of it is.

Reply:
Elaborated in commit 183da45e73336e41c25473b55a33099467cc75ce